File: backend/app/schemas/jobs.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class JobLogResponse(BaseModel):
    id: int
    job_id: int
    original_media_id: int | None
    level: str
    message: str
    file_path: str | None
    details_json: str | None
    created_at: datetime


class JobResponse(BaseModel):
    id: int
    event_id: int | None
    job_type: str
    status: str
    progress_percent: float
    total_items: int
    processed_items: int
    failed_items: int
    started_at: datetime | None
    finished_at: datetime | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime
    logs: list[JobLogResponse] = []


class JobListResponse(BaseModel):
    items: list[JobResponse]

File: desktop/src/types/jobs.ts
export type JobLog = {
  id: number;
  job_id: number;
  original_media_id: number | null;
  level: string;
  message: string;
  file_path: string | null;
  details_json: string | null;
  created_at: string;
};

export type ProcessingJob = {
  id: number;
  event_id: number | null;
  job_type: string;
  status: string;
  progress_percent: number;
  total_items: number;
  processed_items: number;
  failed_items: number;
  started_at: string | null;
  finished_at: string | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
  logs: JobLog[];
};

export type JobListResponse = {
  items: ProcessingJob[];
};
